data/transactions/transaction_test.go

```diff
@@ -51,3 +51,66 @@ func TestTransaction_EstimateEncodedSize(t *testing.T) {
 
 	require.Equal(t, 200, tx.EstimateEncodedSize())
 }
+
+func generateDummyGoNonparticpatingTransaction(addr basics.Address) (tx Transaction) {
+	buf := make([]byte, 10)
+	crypto.RandBytes(buf[:])
+
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+	tx = Transaction{
+		Type: protocol.KeyRegistrationTx,
+		Header: Header{
+			Sender:     addr,
+			Fee:        basics.MicroAlgos{Raw: proto.MinTxnFee},
+			FirstValid: 1,
+			LastValid:  300,
+		},
+		KeyregTxnFields: KeyregTxnFields{Nonparticipation: true},
+	}
+	tx.KeyregTxnFields.VoteFirst = 1
+	tx.KeyregTxnFields.VoteLast = 300
+	tx.KeyregTxnFields.VoteKeyDilution = 1
+
+	tx.KeyregTxnFields.Nonparticipation = true
+	return tx
+}
+
+func TestGoOnlineGoNonparticipatingContradiction(t *testing.T) {
+	// addr has no significance here other than being a normal valid address
+	addr, err := basics.UnmarshalChecksumAddress("NDQCJNNY5WWWFLP4GFZ7MEF2QJSMZYK6OWIV2AQ7OMAVLEFCGGRHFPKJJA")
+	require.NoError(t, err)
+
+	tx := generateDummyGoNonparticpatingTransaction(addr)
+	// Generate keys, they don't need to be good or secure, just present
+	v := crypto.GenerateOneTimeSignatureSecrets(1, 1)
+	// Also generate a new VRF key
+	vrf := crypto.GenerateVRFSecrets()
+	tx.KeyregTxnFields = KeyregTxnFields{
+		VotePK:           v.OneTimeSignatureVerifier,
+		SelectionPK:      vrf.PK,
+		Nonparticipation: true,
+	}
+	// this tx tries to both register keys to go online, and mark an account as non-participating.
+	// it is not well-formed.
+	feeSink := basics.Address{0x7, 0xda, 0xcb, 0x4b, 0x6d, 0x9e, 0xd1, 0x41, 0xb1, 0x75, 0x76, 0xbd, 0x45, 0x9a, 0xe6, 0x42, 0x1d, 0x48, 0x6d, 0xa3, 0xd4, 0xef, 0x22, 0x47, 0xc4, 0x9, 0xa3, 0x96, 0xb8, 0x2e, 0xa2, 0x21}
```

Comment:
Out of curiosity, where did this address come from? Is this the mainnet fee sink?